main.go

```diff
@@ -203,10 +231,31 @@ func main() {
 		os.Exit(1)
 	}
 
+	for _, tp_file := range flagProviders {
+		f, err := os.Open(tp_file)
+		if err != nil {
+			log.Println("couldn't open config file", *flagConfigFile, err)
+			os.Exit(1)
+		}
+		var conf TPConfig
+		dec := yaml.NewDecoder(f)
+		err = dec.Decode(&conf)
+		if err != nil {
```

Comment:
Slightly more idiomatic go:
```suggestion
		if err = dec.Decode(&conf); err != nil {
```